src/tools/run-make-support/src/lib.rs

```diff
@@ -295,6 +295,26 @@ pub fn not_contains<P: AsRef<Path>>(path: P, expected: &str) -> bool {
     !path.as_ref().file_name().is_some_and(|name| name.to_str().unwrap().contains(expected))
 }
 
+/// Builds a static lib (`.lib` on Windows MSVC and `.a` for the rest) with the given name.
+#[track_caller]
+pub fn build_native_static_lib(lib_name: &str) -> PathBuf {
+    let obj_file = if is_msvc() { format!("{lib_name}") } else { format!("{lib_name}.o") };
+    let src = format!("{lib_name}.c");
+    let lib_path = static_lib_name(lib_name);
+    if is_msvc() {
+        cc().arg("-c").out_exe(&obj_file).input(src).run();
+    } else {
+        cc().arg("-v").arg("-c").out_exe(&obj_file).input(src).run();
+    };
+    let mut obj_file = PathBuf::from(format!("{lib_name}.o"));
+    if is_msvc() {
+        obj_file.set_extension("");
+        obj_file.set_extension("obj");
+    }
+    llvm_ar().obj_to_ar().output_input(&lib_path, &obj_file).run();
```

Comment:
Question: With this change, we have *both* `llvm_ar` *and* `ar`-the-crate. See [`ar()`](https://github.com/Oneirical/rust/blob/5d647b26f933afc7ae6f2f8ad48b1dc0b1a23f12/src/tools/run-make-support/src/lib.rs#L65-L76).

I wonder if we could *just* use `llvm_ar`? I'm fine with merging this on a *temporary* basis but would like to either use `llvm_ar` or `ar`-the-crate, but not both, if possible. What do you think?

Reply:
I just removed `ar` and edited the single `rmake.rs` test that was using it.